src/collect/collect.c

```diff
@@ -94,13 +94,17 @@ static void usage(void)
  */
 static int prepare(char *dir, char *filename)
 {
-        char buf[512];
+        char buf[UTIL_PATH_SIZE + 1];
         int r, fd;
 
         r = mkdir(dir, 0700);
         if (r < 0 && errno != EEXIST)
                 return -errno;
 
+        /* Refuse to write to a truncated file path */
+        if (strlen(buf) + 1 + strlen(filename) > sizeof buf - 1)
```

Comment:
I think code style mandates braces when using the sizeof operator

Reply:
OK, thanks!